fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualsProcessingServiceImpl.java

```diff
@@ -1196,10 +1202,13 @@ private void reverseAccrual(LoanTransaction transaction, boolean addEvent) {
     }
 
     private LocalDate getFinalAccrualTransactionDate(Loan loan) {
+        if (configurationDomainService.getAccrualDateConfigForCharge().equals(ACCRUAL_ON_CHARGE_SUBMITTED_ON_DATE)) {
+            return loan.getLastLoanRepaymentScheduleInstallment().getDueDate();
```

Comment:
i wonder whether the charge submitted on date should be the accrual transaction date here, no?